lib/exabgp/bgp/message/open/capability/negotiated.py

```diff
@@ -125,10 +125,16 @@ def validate (self, neighbor):
 		if self.received_open.router_id == RouterID('0.0.0.0'):
 			return (2,3,'0.0.0.0 is an invalid router_id')
 
-		if self.received_open.asn == neighbor.local_as:
```

Comment:
if self.received_open.asn == neighbor.local_as:

self.received_open is the decoded open our peer sent us, so it will have our peer's router-id and ASN
neighbor.local_as, is our ASN, so this check make sure it is an IBGP session.

if self.received_open.router_id == neighbor.router_id:

We are checking that our router-id is not the same their ( unique within the IGP ).